content-src/actions/action-manager.js

```diff
@@ -64,8 +67,8 @@ function RequestBookmarks() {
   return RequestExpect("RECENT_BOOKMARKS_REQUEST", "RECENT_BOOKMARKS_RESPONSE");
 }
 
-function RequestRecentLinks() {
-  return RequestExpect("RECENT_LINKS_REQUEST", "RECENT_LINKS_RESPONSE");
+function RequestRecentLinks(data) {
+  return RequestExpect("RECENT_LINKS_REQUEST", "RECENT_LINKS_RESPONSE", {data: data});
```

Comment:
Kate, note that what you suggest is different from  how Notify is implemented.
https://github.com/mozilla/activity-streams/blob/master/content-src/actions/action-manager.js#L42

Also note that FX wants msg.data not msg.query as in:
https://github.com/mozilla/activity-streams/blob/master/lib/ActivityStreams.js#L80

I am fine either way, but we need to consistent.
Let me know which way you would like to clean the code, I can replace msg.data and msg.query in action handling.